docs/design/proposals/intree-migrate.md

```diff
@@ -0,0 +1,86 @@
+# In-tree storage plugin to CSI Driver Migration
+
+Prior to CSI, Kubernetes provided a powerful volume plugin system. These volume
+plugins were “in-tree” meaning their code was part of the core Kubernetes code
+and shipped with the core Kubernetes binaries. The CSI migration effort enables
+the replacement of existing in-tree storage plugins such as kubernetes.io/rbd
+with a corresponding CSI driver. for more information on CSI migration effort
+refer [design doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-migration.md)
+
+## RBD in-tree plugin to CSI migration
+
+This feature is tracked in Kubernetes for v1.23 as alpha under
+[Kubernetes/enhancements#2963](https://github.com/Kubernetes/enhancements/pull/2963)
+. Below are the changes identified in the design discussions to enable CSI
+migration for RBD plugin:
+
+### ClusterID field in the migration request
+
+Considering the `clusterID` field is a required one for CSI, but in-tree SC
+has `monitors` field as a required thing the clusterID will be sent from the
+migration library based on the monitors field, Kubernetes storage admin supposed
+to create a clusterID based on the monitors hash ( ex: `#md5sum <<<
+"<monaddress:port>"`) in the CSI config map and keep the monitors under this
+configuration before enabling the migration. While CSI driver receive the volume
+ID it will look at the configmap and figure out the mons to do the operations.
+Thus, CSI operations are unchanged or untouched wrt to the clusterID field.
+
+### New Volume Handle/ID for existing volume operations
+
+The existing volume will have a volume handle passed in similar to the following
+format:
+
+```
+mig_mons-<monitors-hash>_image-<imageUUID>_<pool-hash>
+```
+
+Details on the hash values:
+
+* Monitors Hash: this carry a hash value (md5sum) which will be acted as the
+  `clusterID` for the operations in this context.
+
+* ImageUUID: this is the unique UUID generated by Kubernetes for the created
+  volume.
+
+* PoolHash: this is an encoded string of pool name.
+
+The existing in-tree volume's node operations should work without any issues as
+those will be tracked as `static volumes` for the CSI driver. From above volume
+handle, the CSI driver can also connect to the backend cluster and clean/delete
+the image. The migration volume ID carry the information like mons, pool and
+image name which is good enough for the driver to identify and connect to the
+backend cluster for its operations. With above volume handle in place, we will
+be able to support mount, unmount, delete and resize operations.
+
+### Migration secret for CSI operations
+
+Considering taking out admin burden is the main ask from Kubernetes community
+this change is introduced.
+
+The in-tree secret has a data field called "key" which is the base64 admin
```

Comment:
field should be mono-typed, like `key`. That is done in some places, but others incorrectly use double "quotes".